TWLight/applications/models.py

```diff
@@ -403,11 +413,18 @@ def post_revision_commit(sender, instance, **kwargs):
         authorization.authorizer = authorizer
         authorization.partner = instance.partner
 
-        # If this is a proxy partner, set (or reset) the expiry date
+        # If this is a proxy partner, and the proxy_account_length
+        # field is set to false, set (or reset) the expiry date
         # to one year from now
-        if instance.partner.authorization_method == Partner.PROXY:
+        if instance.partner.authorization_method == Partner.PROXY and (instance.partner.proxy_account_length is False or instance.proxy_account_length is None):
```

Comment:
If we've set up the validation for admin correctly, should it ever be the case that `instance.partner.proxy_account_length is False` while authorization method is Proxy?

Reply:
I forgot about this part and I think we agreed we should have a validation because that's never the case. I'll make sure to add it.